csharp/src/SeedLang/Runtime/Events.cs

```diff
@@ -113,4 +113,13 @@ public class EvalEvent : AbstractEvent {
       Value = value;
     }
   }
+
+  // An event which is triggered when an print build-in function is executed.
+  public class PrintEvent : AbstractEvent {
```

Comment:
Do we really need a separate visualizer event for every single built-in function?

From what we see in this code, we already have `EvalEvent` for `eval`, `PrintEvent` for `print`... Will every built-in function require an event finally?

Is it possible to have an event such as `BuiltInFuncCallEvent`, and have the function name and function arguments as its parameters?

Reply:
Two different but related things are being mixed up here.

1. The intrinsic or expected behavior of functions that make side effects to the system (via inputting or outputting): Functions like `print` or `io.stderr.write` or `console.log` are designed to make side effects to the system.

2. The visualizer framework to notify the outer world about what is happening in the SeedLang runtime: e.g., a function call event can be passed out to visualizers to notify the thing that is happening.

I think both two things are valuable to users and it might not be a good choice to mix them up.